R_Scripts/expanded_spectra-trait_reseco_lma_plsr_example.R

```diff
@@ -379,29 +382,39 @@ head(val.plsr.output)
 # JK regression coefficient plot
 f.plot.coef(Z = t(Jackknife_coef), wv = seq(Start.wave,End.wave,1), 
             plot_label="Jackknife regression coefficients",position = 'bottomleft')
+png(file=file.path(outdir,paste0(inVar,'_jackknife_regression_coefficients.png')),
+    height=2100,width=3800, res=340)
+f.plot.coef(Z = t(Jackknife_coef), wv = seq(Start.wave,End.wave,1), 
+            plot_label="Jackknife regression coefficients",position = 'bottomleft')
+abline(h=0,lty=2,col="grey50")
+box(lwd=2.2)
+dev.off()
 
 # JK validation plot
+RMSEP <- sqrt(mean(val.plsr.output$PLSR_Residuals^2))
+pecr_RMSEP <- RMSEP/mean(val.plsr.output[,inVar])*100
+r2 <- round(pls::R2(plsr.out, newdata = val.plsr.data)$val[nComps+1],2)
+expr <- vector("expression", 3)
+expr[[1]] <- bquote(R^2==.(r2))
+expr[[2]] <- bquote(RMSEP==.(round(RMSEP,2)))
+expr[[3]] <- bquote("%RMSEP"==.(round(pecr_RMSEP,2)))
 rng_vals <- c(min(val.plsr.output$LPI), max(val.plsr.output$UPI))
-jk_val_scatterplot <- ggplot(val.plsr.output, aes(x=PLSR_Predicted, y=get(inVar))) + 
-  theme_bw()+ geom_errorbar(aes(xmin = LPI,xmax = UPI),color='grey',width=0.2) + 
-  geom_errorbar(aes(xmin = LCI,xmax = UCI),color='blue',width=0.2)+ geom_point(size=1.3)  + 
-  geom_abline(intercept = 0, slope = 1, color="grey30", 
-              linetype="dashed", size=0.7) + 
-  xlim(rng_vals[1], rng_vals[2]) + 
-  ylim(rng_vals[1], rng_vals[2]) +
-  labs(x=paste0("Predicted ", paste(inVar), " (units)"),
-       y=paste0("Observed ", paste(inVar), " (units)"),
-       title=paste0("Confidence Interval: Blue;   Prediction_Interval: grey")) + 
-  theme(axis.text=element_text(size=18),legend.position = 'right',
-        axis.title=element_text(size=20, face="bold"), 
-        axis.text.x = element_text(angle = 0,vjust = 0.5),
-        panel.border = element_rect(linetype = "solid", fill = NA, size=1.5))
-print(jk_val_scatterplot)
-ggsave(paste0(inVar,"_PLSR_validation_scatterplot.png"), plot = jk_val_scatterplot, device="png",
-              width = 32, height = 30, units = "cm", dpi = 300)
+par(mfrow=c(1,1), mar=c(4.2,5.3,1,0.4), oma=c(0, 0.1, 0, 0.2))
+plotCI(val.plsr.output$PLSR_Predicted,val.plsr.output[,inVar], 
+       li=val.plsr.output$LPI, ui=val.plsr.output$UPI, gap=0.009,sfrac=0.004, 
+       lwd=1.6, xlim=c(rng_vals[1], rng_vals[2]), ylim=c(rng_vals[1], rng_vals[2]), 
+       err="x", pch=21, col="black", pt.bg=alpha("grey70",0.7), scol="grey50",
+       cex=2, xlab=paste0("Predicted ", paste(inVar), " (units)"),
+       ylab=paste0("Observed ", paste(inVar), " (units)"),
+       cex.axis=1.5,cex.lab=1.8)
+abline(0,1,lty=2,lw=2)
+legend("topleft", legend=expr, bty="n", cex=1.5)
+box(lwd=2.2)
+dev.copy(png,file.path(outdir,paste0(inVar,"_PLSR_validation_scatterplot.png")), 
```

Comment:
@JulienLamour @neo0351 added this so we can print to file and screen in the scripts